src/scaffoldmaker/meshtypes/meshtype_3d_brainstem.py

```diff
@@ -0,0 +1,355 @@
+                    elemsPerColumn = [elementsCountAcrossMinor-2]+ [elementsCountAcrossMinor]*(elementsCountAcrossMajor-2)+[elementsCountAcrossMinor-2]
+                    cumElemsPerColumn = [0] + [sum(elemsPerColumn[:i + 1]) for i in range(len(elemsPerColumn))]
+                    for i in range(3):
+                        prow[i] -= 0.1 if prow[i] == 1 else 0
+                    [nx, ny, nz] = [int(prow[0] * elementsCountAcrossMajor),
+                                    int(prow[1] * elementsCountAcrossMinor),
+                                    int(prow[2] * elementsCountAlong)]
+                    if (nx == 0 or nx >= elementsCountAcrossMajor-1):# and ny > (elementsCountAcrossMinor/2):
+                        ny = int(ny/2)
+                    nx -= 1 if nx == elementsCountAcrossMajor else 0
+                    nz -= 1 if nz == elementsCountAlong else 0
+                    elementIndex[nuc].append(ny + cumElemsPerColumn[nx] + (elemPerLayer*nz) + 1)
+                    xi1[nuc] = 1 if prow == 1 else prow
+
+            # order the same as nucAbbrev
+            annotationGroups = [ NAGroup, NTSGroup, rapheGroup, mPBGroup, lPBGroup, KFNGroup, RTNGroup, botzGroup, prebotzGroup, rVRGGroup, cVRGGroup, DRGGroup ]
+        else:
+            annotationGroups = []
+        brainstemGroup = AnnotationGroup(region, get_brainstem_annotation_term('brainstem'))
+        midbrainGroup = AnnotationGroup(region, get_brainstem_annotation_term('midbrain'))
+        ponsGroup = AnnotationGroup(region, get_brainstem_annotation_term('pons'))
+        medullaGroup = AnnotationGroup(region, get_brainstem_annotation_term('medulla oblongata'))
+        brainstemMeshGroup = brainstemGroup.getMeshGroup(mesh)
+        midbrainMeshGroup = midbrainGroup.getMeshGroup(mesh)
+        ponsMeshGroup = ponsGroup.getMeshGroup(mesh)
+        medullaMeshGroup = medullaGroup.getMeshGroup(mesh)
+        allMeshGroups = [igroup.getMeshGroup(mesh) for igroup in annotationGroups]
+        annotationGroups += [brainstemGroup, midbrainGroup, ponsGroup, medullaGroup]
+
+        ############################################################
+        # point markers for alignment during scaffoldfitter
+        ############################################################
+        eIndexPM = {}
+        xiPM = {}
+        pointMarkers = {}
+        eIndexPM['caudal-dorsal'] = int(elemPerLayer/2)
+        eIndexPM['midRostCaud-dorsal'] = int(elemPerLayer / 2) + (elemPerLayer * int((elementsCountAlong/2)-1))
+        eIndexPM['rostral-dorsal'] = (elemPerLayer*(elementsCountAlong-1)) + int(elemPerLayer/2)
+        eIndexPM['caudal-ventral'] = int(elemPerLayer/2) - (elementsCountAcrossMinor-1)
+        eIndexPM['midRostCaud-ventral'] = eIndexPM['midRostCaud-dorsal'] - (elementsCountAcrossMinor-1)
+        eIndexPM['rostral-ventral'] = int((elemPerLayer*(elementsCountAlong-1)) + (int(elemPerLayer/2) - elementsCountAcrossMinor + 1))
```

Comment:
We're now adding the marker points as annotation groups and putting the points in not just the marker group, but the brainstem group and their own annotation group.
We will add the point annotations with the name and term Id "None" (a string with the word None in it) and request the terms from SciCrunch using the google sheet the mappers maintain.
All the names need to be qualified with "brainstem" and point/junction so they are clear what anatomy they are for in SciCrunch. Since I can't run it yet, I can't comment any further on the locations. Avoid abbreviations like midRostCaud.